cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerParcelManagementService.java

```diff
@@ -60,10 +61,19 @@ public Set<ParcelInfo> getParcelsInStatus(ParcelsResourceApi parcelsResourceApi,
     public Set<ParcelInfo> getAllParcels(ParcelsResourceApi parcelsResourceApi, String stackName) throws ApiException {
         LOGGER.debug("Retrieving all available parcels from CM");
         return getClouderaManagerParcels(parcelsResourceApi, stackName).getItems().stream()
-                .map(apiParcel -> new ParcelInfo(apiParcel.getProduct(), apiParcel.getVersion(), ParcelStatus.valueOf(apiParcel.getStage())))
+                .map(apiParcel -> new ParcelInfo(apiParcel.getProduct(), apiParcel.getVersion(), getParcelStatus(apiParcel)))
                 .collect(Collectors.toSet());
     }
 
+    private ParcelStatus getParcelStatus(ApiParcel apiParcel) {
+        try {
+            return ParcelStatus.valueOf(apiParcel.getStage());
+        } catch (IllegalArgumentException e) {
+            LOGGER.warn("An unknown parcel status occurred: {} for parcel {}. Returning UNKNOWN state.", apiParcel.getStage(), apiParcel);
```

Comment:
I would add the exception for the logger, it would log the stack trace which can be helpful